hnn_core/optimization.py

```diff
@@ -480,8 +483,12 @@ def optimize_evoked(net, tstop, n_trials, target_dpl, initial_dpl, maxiter=50,
 
     net = net.copy()
 
-    drive_names = [key for key in net.external_drives.keys()
-                   if net.external_drives[key]['type'] == 'evoked']
+    if which_drives == 'all':
+        drive_names = [key for key in net.external_drives.keys()
+                       if net.external_drives[key]['type'] == 'evoked']
+    else:
+        drive_names = [key for key in net.external_drives.keys()
+                       if net.external_drives[key]['type'] == 'evoked' and key in which_drives]
```

Comment:
This looks good @mohdsherif! My only recommendation would be to throw a `ValueError` if the items in `which_drives` are not evoked drive names defined in `net.external_drives[key]['type']` so that the user doesn't accidentally define a drive that is silently ignored during optimization.

Then, you should add a test that raises this error. You can probably just add such a test [here](https://github.com/jonescompneurolab/hnn-core/blob/b974fed4317b9b03ec732850829d9c17841c6ee2/hnn_core/tests/test_optimization.py#L112) to test for all expected user errors that might emerge when `which_drives` is `'all'` and when it is of type list.